hooks/useFilter.js

```diff
@@ -140,6 +140,40 @@ export const getCategoryLinkAttributes = (
   };
 };
 
+export const getCampaignLinkAttributesFromRouter = (router, options = {}) => {
+  return getCampaignLinkAttributes(
+    get(router, "query.campaign"),
+    arrayToPath(get(router, "query.parameters", [])),
+    router.query,
+    options
+  );
+};
+
+export const getCampaignLinkAttributes = (
+  campaign,
+  parameters = "",
+  query = {},
+  options = {}
+) => {
+  const newQuery = get(options, "toDelete")
+    ? queryToQuery(
+        query,
```

Comment:
no ak to spravne chapem, tak kampane si spravil iba do euronicsu. Preco potom fetche a podporne metody su v generalnej libke?